sql/core/src/main/scala/org/apache/spark/sql/execution/metric/CustomMetrics.scala

```diff
@@ -51,7 +51,7 @@ object CustomMetrics {
       currentMetricsValues: Seq[CustomTaskMetric],
       customMetrics: Map[String, SQLMetric]): Unit = {
     currentMetricsValues.foreach { metric =>
-      customMetrics(metric.name()).set(metric.value())
+      customMetrics.get(metric.name()).map(_.set(metric.value()))
```

Comment:
This looks more robust. Could you add a test case for this no-op too?

Reply:
Also, it would be great if you put the explanation at line 48.

Reply:
Ok.

Reply:
Added test and updated the method doc. Thanks.